components/at_mention/at_mention.jsx

```diff
@@ -71,8 +72,36 @@ export default class AtMention extends React.PureComponent {
         return '';
     }
 
+    getGroupFromMentionName(props) {
+        const groupsByName = props.groupsByName;
+        let mentionName = props.mentionName.toLowerCase();
+
+        while (mentionName.length > 0) {
+            if (groupsByName.hasOwnProperty(mentionName)) {
+                return groupsByName[mentionName];
+            }
+
+            // Repeatedly trim off trailing punctuation in case this is at the end of a sentence
+            if ((/[._-]$/).test(mentionName)) {
+                mentionName = mentionName.substring(0, mentionName.length - 1);
+            } else {
+                break;
+            }
+        }
+
+        return {};
+    }
+
     render() {
         const user = this.getUserFromMentionName(this.props);
+
+        if (!user) {
+            const group = this.getGroupFromMentionName(this.props);
+            if (group.allow_reference) {
```

Comment:
I'm pretty sure you don't need this check, but prob. safer to keep it, if things change in the future.